### DISCUSSION TITLE
Where should OCaml-only unit tests go?

### DISCUSSION THREAD

**Discussion by ahelwer** [tlaplus/tlapm#142] [Q&A] [answered]
Looking through the `test` directory, the tests all seem to be .tla files which are run through the TLAPM CLI using bash scripts in the `test/TOOLS` directory. Are there any unit tests that are written in OCaml that only test OCaml functions without otherwise touching the filesystem? If I wanted to write one of those, where should it go and how should it hook into the rest of the test system?

**Comment by kape1395** [comment, marked as answer]
Plain ocaml unit tests are now implemented using ppx. E.g. https://github.com/tlaplus/tlapm/blob/117b5cb1fc0734a5667d6c415cf0a9aee978792b/src/module/m_save.ml#L319
The dune file then has to have inline tests enabled andthe corresponding preprocessor defined, e.g. https://github.com/tlaplus/tlapm/blob/117b5cb1fc0734a5667d6c415cf0a9aee978792b/src/dune#L19

More on tests in dune: https://dune.readthedocs.io/en/stable/tests.html

**Reply by ahelwer** [reply]
So you can run that test in isolation with `dune exec src/module/m_save.ml`?

**Reply by ahelwer** [reply]
Ah, the correct command is `dune runtest src`.
